service/dap/daptest/client.go

```diff
@@ -120,6 +118,16 @@ func (c *Client) ExpectConfigurationDoneResponse(t *testing.T) *dap.Configuratio
 	return c.expectReadProtocolMessage(t).(*dap.ConfigurationDoneResponse)
 }
 
+func (c *Client) ExpectThreadsResponse(t *testing.T) *dap.ThreadsResponse {
+	t.Helper()
+	return c.expectReadProtocolMessage(t).(*dap.ThreadsResponse)
```

Comment:
if expectReadProtoMessage returns a nil or a message that can't be *dap.ThreadsResponse, this type assertion will trigger runtime panic. In order to avoid, you can use

r, ok := c.expectReadProtocolMessage(t).(*dap.ThreadsResponse)
and if !ok, t.Errorf, otherwise return r.